render/gles2/renderer.c

```diff
@@ -25,7 +25,6 @@ static struct wlr_gles2_renderer *gles2_get_renderer(
 static struct wlr_gles2_renderer *gles2_get_renderer_in_context(
 		struct wlr_renderer *wlr_renderer) {
 	struct wlr_gles2_renderer *renderer = gles2_get_renderer(wlr_renderer);
-	assert(wlr_egl_is_current(renderer->egl));
```

Comment:
We still need to assert this for functions called between `wlr_renderer_begin` and `wlr_renderer_end`. I'd suggest keeping this function as is and adding a new `gles2_get_renderer_switch_context` for textures.